### PR TITLE
Provide a sendfile implementation for Netty based on io_uring.

### PR DESCRIPTION
**Pull request by dreamlike-ocean**
Motivation:

Provide a sendfile implementation for Netty based on io_uring.

https://github.com/netty/netty-incubator-transport-io_uring/issues/210

Modification:

Provide a sendfile implementation for Netty based on io_uring.

Result:

https://github.com/netty/netty-incubator-transport-io_uring/issues/210

### CODE REVIEW

File: transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUring.java [hunk 1]
```diff
@@ -41,7 +43,8 @@ public final class IoUring {
                     try {
                         ringBuffer = Native.createRingBuffer();
                         Native.checkAllIOSupported(ringBuffer.fd());
-                        supported = Native.isIOUringCqeFSockNonEmptySupported(ringBuffer.fd());
+                        socketNonEmptySupported = Native.isIOUringCqeFSockNonEmptySupported(ringBuffer.fd());
+                        spliceSupported = Native.isIOUringCqeFSockNonEmptySupported(ringBuffer.fd());
```
Comment (franz1981) [inline transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUring.java:47]: replace with `Native.isIOUringSpliceSupported`

Comment (dreamlike-ocean) [reply transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUring.java:47]: > replace with 'isIOUringSpliceSupported`

I get it. Here is a mistake of mine.

**Comment by normanmaurer** [comment]
Will have a look soon

**Comment by dreamlike-ocean** [comment]
> @dreamlike-ocean did you sign our ICLA? https://netty.io/s/icla

Yes. I have signed the ICLA.

**Comment by normanmaurer** [comment]
@dreamlike-ocean ok ... I will take it from here. This needs a bit of work

**Pull request closed without merging**